### PR TITLE
Add webpush notifications

### PR DESCRIPTION
**Pull request by ProCode2**
Adding web push notifications for various actions on the site.

- [x] Configure Workbox to use custom ServiceWorker code
- [x] Show notifications when triggered
- [x] handle multi-device subscription logic
- [x] handle notification click
- [x] Resolve to correct page on notification click.

https://www.loom.com/share/fc048c43e5544b49886723bdd7737083

### CODE REVIEW

**Comment by Apurva-Nagar** [comment]
@bodhish 

- [x] Configure Workbox to use custom ServiceWorker code
- [x] Show notifications when triggered (with hardcoded information for now)
- [x] handle multi-device subscription logic
- [x] handle notification click (with hardcoded information for now)
- [x] Resolve to correct page on notification click.

File: src/Components/Notifications/ShowPushNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const [isLoading, setIsLoading] = useState(true);
+  const { data }: any = props;
+  const getNotificationData = async () => {
+    const res = await axios.get(`/api/v1/notification/${data.id}/`, {
+      headers: {
+        Authorization: `Bearer ${localStorage.getItem("care_access_token")}`,
+      },
+    });
+    return res.data;
+  };
```
Comment (bodhish) [inline src/Components/Notifications/ShowPushNotification.tsx]: Can we use the `fire_request` Helper

File: src/Components/Notifications/NotificationsList.tsx [hunk 2]
```diff
@@ -27,8 +33,126 @@ export default function ResultList() {
...
+        pf_endpoint: push.endpoint,
+        pf_p256dh: p256dh,
+        pf_auth: auth,
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${localStorage.getItem("care_access_token")}`,
+        },
+      }
+    );
```
Comment (bodhish) [inline src/Components/Notifications/NotificationsList.tsx]: Can we use the `fire_request` Helper

File: src/Components/Notifications/NotificationsList.tsx [hunk 3]
```diff
@@ -27,8 +33,126 @@ export default function ResultList() {
...
+                pf_auth: "",
+              },
+              {
+                headers: {
+                  Authorization: `Bearer ${localStorage.getItem(
+                    "care_access_token"
+                  )}`,
+                },
+              }
+            );
```
Comment (bodhish) [inline src/Components/Notifications/NotificationsList.tsx]: Can we use the `fire_request` Helper

File: src/Components/Notifications/NotificationsList.tsx [hunk 4]
```diff
@@ -27,8 +33,126 @@ export default function ResultList() {
...
 
+  const [isSubscribed, setIsSubscribed] = useState("");
+  const [isSubscribing, setIsSubscribing] = useState(false);
+
+  const intialSubscriptionState = async () => {
+    const res = await axios.get(`/api/v1/users/${username}/pnconfig/`, {
+      headers: {
+        Authorization: `Bearer ${localStorage.getItem("care_access_token")}`,
+      },
+    });
```
Comment (bodhish) [inline src/Components/Notifications/NotificationsList.tsx]: Can we use the `fire_request` Helper

File: webpack.config.js [hunk 5]
```diff
@@ -10,14 +10,13 @@ const OptimizeCssAssetsPlugin = require("optimize-css-assets-webpack-plugin");
 const safePostCssParser = require("postcss-safe-parser");
 
 const prodPlugins = (isDev) => {
-  if (isDev) {
-    return [];
-  }
+  // if (isDev) {
+  //   return [];
+  // }
```
Comment (bodhish) [inline webpack.config.js]: Can we disable push notification in development for better performance while compiling?

**Review by bodhish — CHANGES_REQUESTED** [review]

**Review by bodhish — APPROVED** [review]
Nice 🔥

**Review by bodhish — APPROVED** [review]

**Pull request merged**